### PR TITLE
Sovereign shard interceptors container factory 

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Sovereign shard interceptors container factory
## Proposed changes
- Implemented it + injected factories

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: testscommon/sovereign/incomgHeaderSubscriberStub.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package sovereign
+
+import "github.com/multiversx/mx-chain-core-go/data/sovereign"
+
+type IncomingHeaderSubscriberStub struct {
+	AddHeaderCalled func(headerHash []byte, header sovereign.IncomingHeaderHandler) error
+}
+
+func (ihs *IncomingHeaderSubscriberStub) AddHeader(headerHash []byte, header sovereign.IncomingHeaderHandler) error {
```
Comment (sasurobert) [inline testscommon/sovereign/incomgHeaderSubscriberStub.go:9]: missing comments.

File: process/interceptors/processor/sovereignHdrInterceptorProcessor.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+type ArgsSovereignHeaderInterceptorProcessor struct {
+	BlockBlackList           process.TimeCacher
+	Hasher                   hashing.Hasher
+	Marshaller               marshal.Marshalizer
+	IncomingHeaderSubscriber process.IncomingHeaderSubscriber
+}
+
+type sovereignHeaderInterceptorProcessor struct {
+	blackList                process.TimeCacher
+	Hasher                   hashing.Hasher
```
Comment (SebastianMarian) [inline process/interceptors/processor/sovereignHdrInterceptorProcessor.go:23]: We need these 3 variables to be exported?

File: process/interceptors/processor/sovereignHdrInterceptorProcessor.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+// Validate checks if the intercepted data can be processed
+func (hip *sovereignHeaderInterceptorProcessor) Validate(data process.InterceptedData, _ core.PeerID) error {
+	_, ok := data.(process.ExtendedHeaderValidatorHandler)
+	if !ok {
+		return fmt.Errorf("sovereignHeaderInterceptorProcessor.Validate error: %w", process.ErrWrongTypeAssertion)
+	}
+
+	return nil
```
Comment (SebastianMarian) [inline process/interceptors/processor/sovereignHdrInterceptorProcessor.go:52]: Here should be added more code for validation? If so, you can put a TODO here!

File: process/interceptors/processor/sovereignHdrInterceptorProcessor.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+// Save will save the received data into headers pool
+func (hip *sovereignHeaderInterceptorProcessor) Save(data process.InterceptedData, _ core.PeerID, _ string) error {
+	_, ok := data.(process.ExtendedHeaderValidatorHandler)
+	if !ok {
+		return fmt.Errorf("sovereignHeaderInterceptorProcessor.Save error: %w", process.ErrWrongTypeAssertion)
+	}
+
+	return nil
```
Comment (SebastianMarian) [inline process/interceptors/processor/sovereignHdrInterceptorProcessor.go:62]: Here should be added more code for saving, right ? If so, you can put a TODO here!

File: process/interface.go [hunk 5]
```diff
@@ -113,6 +114,12 @@ type HdrValidatorHandler interface {
 	HeaderHandler() data.HeaderHandler
 }
 
+// ExtendedHeaderValidatorHandler defines extended header validator functionality
+type ExtendedHeaderValidatorHandler interface {
+	Hash() []byte
+	GetExtendedHeader() data.ShardHeaderExtendedHandler
```
Comment (SebastianMarian) [inline process/interface.go:120]: IsInterfaceNil() ?

File: testscommon/sovereign/incomgHeaderSubscriberStub.go (hunk 1 above)
Comment (SebastianMarian) [reply testscommon/sovereign/incomgHeaderSubscriberStub.go:9]: +1

Comment (mariusmihaic) [reply testscommon/sovereign/incomgHeaderSubscriberStub.go:9]: Right, added them

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**